modules/ROOT/pages/batch-phases.adoc

```diff
@@ -0,0 +1,135 @@
++
+[[batchJobInstanceId]]
+Within the scope of the Batch Job component, Mule exposes each batch job instance through the `batchJobInstanceId` variable. You can use `vars.batchJobInstanceId` to access the identifier for an instance in any batch processing phase. The auto-generated identifier is a UUID, but you can change it to a more readable name. For details, see xref:batch-job-instance-id.adoc[].
++
+. Mule automatically uses DataWeave to split the input to the Batch Job component into records and stores the records in a persistent queue. The input is a serialized message payload that Mule splits into a collection of records. During the <<phase_process, Process phase>>, a Batch Step component pulls records from this queue.
++
+This activity is "all or nothing," which means that Mule either successfully generates and queues a record for every item from the input, or the entire event fails with an error type, such as `MULE:EXPRESSION` message, and an error message, such as `"Expecting Array or Object but got String." evaluating expression: "payload".`
++
+In Anypoint Runtime Manager, the queue name is prefixed with `BSQ`, for example:
++
+image:mruntime-batch-bsq.png[Batch step queue transaction log]
++
+. Mule presents the batch job instance and queue of records to its first Batch Step component for processing.
+
+Each operation within this phase blocks other execution within the flow until it completes. The logs identify them as `BLOCKING` operations. After this phase completes, the flow continues to execute without waiting for batch processing operations in the next phase to execute or finish processing records they receive.
+
+[[phase_process]]
+== Process Phase
+
+The Process phase begins after the Load and Dispatch phase finishes loading records to the queue and starts execution of a batch job instance. In this phase, Mule pulls records from the queue to build record blocks of the configured batch block size (100 records per block, by default). Then Mule sends the blocks to the Batch Step component.
+
+Within the Batch Step component, processing takes place in parallel at the block level, but records within each block process sequentially by default. Mule uses its auto-tuning capabilities to determine how many threads to use and the level of parallelism to apply (see xref:execution-engine.adoc[Execution Engine]).
+
+During this phase, processors within the components can access and modify records using the Mule variables `payload` and `vars`, but _not_ `attributes` because the Batch Job component ignores Mule attributes and does not make them available to its child components. The Batch Job component consumes records after all record processing finishes. So any transmission of _processed_ records to an external server or service must take place _within_ the Batch Step or Batch Aggregator components. For Mule variable propagation rules, see xref:batch-processing-concept.adoc#variable_propagation[Variable Propagation].
```

Comment:
TODO: maybe we remove the entire paragraph, since all is discussed elsewhere. An advantage of having it here is that it summarizes everything that is detailed later. TBD

Reply:
Ivan: when create the records, not including the attributes, which happens in the l&d phase.

Reply:
remove ", but _not_ `attributes` because the Batch Job component ignores Mule attributes and does not make them available to its child components"

Reply:
DONE